astropy/io/fits/hdu/hdulist.py

```diff
@@ -1057,7 +1065,8 @@ def _readfrom(cls, fileobj=None, data=None, mode=None, memmap=None,
         if fileobj is not None:
             if not isinstance(fileobj, _File):
                 # instantiate a FITS file object (ffo)
-                fileobj = _File(fileobj, mode=mode, memmap=memmap, cache=cache)
+                fileobj = _File(fileobj, mode=mode, memmap=memmap,
+                                cache=cache, **kwargs)
```

Comment:
Here it would be very nice to explicitly add the new parameters rather than hiding them in kwargs.

Reply:
I agree.  Extra discussion on this topic is captured here: https://github.com/astropy/astropy/pull/13238#discussion_r934803704

Reply:
Update: I have now made the new parameters explicit in response to this feedback. (commit 6e7b6095f97e25d8491cd2ae0d61b2052dc19d3c)